### PR TITLE
[NLP-181] Read backend and nginx port from configurable environment variables

### PR DESCRIPTION
**Pull request by boyangdong00**
- set environment variable in docker-compose and nginx.conf

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,9 +26,15 @@ services:
...
+      - NGINX_PORT=${NGINX_PORT}
+      - BACKEND_PORT=${BACKEND_PORT}
+      - DOLLAR=${DOLLAR}
     ports:
-      - "80:80"
-    restart: always
+      - "${NGINX_PORT}:${NGINX_PORT}"
+    expose:
+      - "${NGINX_PORT}"
+    command: /bin/bash -c "envsubst < /etc/nginx/templates/form.template > /etc/nginx/conf.d/default.conf && cat /etc/nginx/conf.d/default.conf && find /etc/nginx -print && nginx -g 'daemon off;'"
```
Comment (JCayabyab) [inline docker-compose.yml:37]: You can get rid of the `cat` and `find` calls in this pipe, they were only for debugging.

File: docker-compose.yml [hunk 2]
```diff
@@ -4,17 +4,19 @@ services:
     container_name: backend
     build:
       context: .
-      dockerfile: Dockerfile
+      dockerfile: Dockerfile
```
Comment (JCayabyab) [inline docker-compose.yml:7]: Remove this unneeded whitespace (unfortunately we don't have a linter for this .yml file 😢 )

**Review by JCayabyab — APPROVED** [review]
Small changes. Please only merge after addressing these changes, and ping me when these are merged in.

File: docker-compose.yml (hunk 2 above)
Comment (boyangdong00) [reply docker-compose.yml]: done

File: docker-compose.yml (hunk 1 above)
Comment (boyangdong00) [reply docker-compose.yml]: done

**Pull request merged**